backend/src/process.py

```diff
@@ -181,7 +181,8 @@ async def __process(self, node: UsableData) -> Any:
             output = await self.loop.run_in_executor(None, run_func)
             node_outputs = node_instance.get_outputs()
             broadcast_data: Dict[int, Any] = dict()
-            if len(node_outputs) > 0:
+            # Only broadcast the output if the node has outputs and the output is not cached
+            if len(node_outputs) > 0 and self.output_cache.get(node_id, None) is None:
```

Comment:
This is so it doesn't broadcast when you run the chain. Without it, it gets from the cache and the broadcasts it again even though it doesn't need to, so you see the node flicker for a short while

Reply:
Then that's a frontend bug. The backend should be allowed to repeat broadcasts for nodes. Child nodes have problems otherwise.

I would suggest doing a deep equal on received broadcast data to figure out whether the broadcast data map needs to be updated at all.

Reply:
How would child nodes have problems?

Reply:
The backend doesn't cache anything in the main cache for iterator children, if that's what you mean. Those get treated as an entirely new executor, remember?

I don't disagree I should do a deep equals check when setting stuff though, but I also think we shouldn't be wasting IPC time sending data we don't need to send.